packages/wallets/src/features/cashier/modules/Transfer/hooks/useWalletTransfer.ts

```diff
@@ -0,0 +1,200 @@
+import { useMemo } from 'react';
+import { useActiveWalletAccount, useAuthorize, useCurrencyConfig, useTransferAccounts } from '@deriv/api';
+import { displayMoney } from '@deriv/api/src/utils/display-money';
+import { getAccountName } from '../helpers';
+
+type TAccountType = NonNullable<
+    NonNullable<ReturnType<typeof useTransferAccounts>['data']>['accounts']
+>[number]['account_type'];
+type TMt5MarketType = ReturnType<typeof getMarketType>;
+
+const useWalletTransfer = (fromAccountLoginId?: string) => {
+    const { data: authorizeData, isLoading: isAuthorizeLoading } = useAuthorize();
+    const { data: activeWallet, isLoading: isActiveWalletLoading } = useActiveWalletAccount();
+    const { getConfig, isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+    const { data, isFetchedAfterMount: isTransferAccountsLoading } = useTransferAccounts();
+
+    const isLoading =
+        isAuthorizeLoading || !isTransferAccountsLoading || isCurrencyConfigLoading || isActiveWalletLoading;
+
+    const modifiedTransferAccounts = useMemo(() => {
```

Comment:
what does it mean "modifiedTransferAccount" ? is this new type of data / new type of account structure?

Reply:
cc @markwylde-deriv - it looks very much like a topic for data modelling, otherwise, we gonna have dozens of different "Account" objects,

Reply:
hey @wojciech-deriv 
data, which we receive from the BE is not enough to proper render` transfer account item` UI (in transfer account item we should show `icon`, `landing company`, `balance` in proper format, etc.) 
actually we are not modifying data from response, we are adding new properties (that help us render the ui) based on received data and extend them
<img width="428" alt="Screenshot 2023-11-08 at 12 55 14" src="https://github.com/binary-com/deriv-app/assets/103181646/273b7834-295e-493f-beeb-51ec2b50f939">

Reply:
Do we have a useAccounts hook, or something that can be shared?

I think, for now, the deriv/api hooks appear to be how we're hydrating the server response with additional logic we need for now. Soon, we can hopefully put this in an SDK layer and/or eventually get this running server side.

Reply:
hey @markwylde-deriv 
yes we have smth like useAccountsList hook in api package
IMHO this is good to have smth like one source like useAccountsList which has all attributes we are using across wallets that we can easily render the UI:

1. Icon of the app
2. Proper balance display
3. Landing company
4. Icon for the current active wallet
5. Account name (necessarily)